Source/RpmAvatarCreator/Public/NARF/Elements/RpmAssetSelectionWidget.h

```diff
@@ -0,0 +1,55 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "RpmAvatarCreatorTypes.h"
+#include "Blueprint/UserWidget.h"
+#include "RpmAssetSelectionWidget.generated.h"
+
+class URpmAssetButton;
+class UImage;
+
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FAssetSelectedEvent, const FRpmPartnerAsset&, AssetData);
+UCLASS()
+class RPMAVATARCREATOR_API URpmAssetSelectionWidget : public UUserWidget
+{
+	GENERATED_BODY()
+
+public:
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Ready Player Me/Creator")
+	TSubclassOf<URpmAssetButton> ButtonBlueprint;
+
+	UPROPERTY(BlueprintAssignable, Category = "Events")
+	FAssetSelectedEvent OnAssetSelected;
+
+	UFUNCTION(BlueprintCallable, Category = "Ready Player Me/Creator")
+	void LoadAndCreateButtons(const EAvatarGender& Gender);
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Ready Player Me/Creator")
+	EAvatarBodyType AvatarBodyType;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Ready Player Me/Creator")
+	ERpmPartnerAssetType AssetType;
+	
+	UFUNCTION()
+	void AssetIconFetched(const FRpmPartnerAsset& PartnerAsset);
+
+protected:
+	virtual void NativeConstruct() override;
+
+private:
+	UPROPERTY(meta = (BindWidget))
+	UPanelWidget* ButtonContainer;
+	
+	TMap<FRpmPartnerAsset, URpmAssetButton*> AssetButtonMap;
+	URpmAssetButton* CurrentlySelectedButton;
+	TSharedPtr<class RpmAssetFetcher> AssetFetcher;
```

Comment:
I assume you copied the asset fetcher class from Unity and pasted it in Unreal. This will not work. Assets are fetched before the editor is opened. and for fetching them RpmAvatarCreatorApi is used.